### PR TITLE
feat: add a debug page with basic information about split store

### PR DESCRIPTION
**Pull request by jancionear**
Add a new debug page: `/debug/pages/split_store`
This page provides basic information about the state of split store.
An example report looks like this:

image

Fixes: https://github.com/near/nearcore/issues/9549

### CODE REVIEW

**Comment by jancionear** [comment]
Tbf I'm not entirely sure about this solution. I still don't understand what the cold store does, why is it even needed? I looked around a bit but couldn't find a clear explanation.
But I put something together, I think it fixes #9549 as the issue only asked to display the cold store head.
I'd appreciate feedback.

**Comment by akhi3030** [comment]
Looks like wacban is on top of this.  Removing myself from reviewers.

**Comment by wacban** [comment]
I can't find good documentation on that topic but perhaps @posvyatokum or @andrei-near will know. 
TLDR 
- regular nodes only store the most recent chain history
- archival nodes store full chain history
- legacy archival nodes store everything in a single RocksDB, on a single, huge and expensive SSD disk
- split store archival nodes store the most recent history and the rest of the history on two separate disks. 
- the hot disk is typically SSD, it's nice and fast, the cold disk is typically HDD, it's big and slow but cheap
- benefit 1 - hot being small, is much faster and thanks to this the node can keep up with the network
- benefit 2 - massive cost saving thanks to HDD being much cheaper, especially for such large volumes

File: chain/jsonrpc-primitives/src/types/split_storage.rs [hunk 1]
```diff
@@ -39,3 +41,16 @@ impl From<RpcSplitStorageInfoError> for crate::errors::RpcError {
         Self::new_internal_or_handler_error(error_data, error_data_value)
     }
 }
+
+impl RpcSplitStorageInfoError {
+    // Implementing From<RpcSplitStorageInfoError> for RpcStatusError causes cargo to spit out hundreds
+    // of lines of compilation errors. I don't want to spend time debugging this, so let's use this function instead.
+    // It's good enough.
```
Comment (wacban) [inline chain/jsonrpc-primitives/src/types/split_storage.rs:48]: lol
if you want we can try together, for the sake of learing rust but as you said it's just fine as is

**Review by wacban — APPROVED** [review]
LGTM

File: chain/jsonrpc-primitives/src/types/split_storage.rs (hunk 1 above)
Comment (jancionear) [reply chain/jsonrpc-primitives/src/types/split_storage.rs:48]: Eh I'm fine with leaving it as is. The errors looked pretty intimidating, and there is no point in spending mental energy on trying to figure out what it is complaining about. "Focus on what matters", right? :P

Comment (jancionear) [reply chain/jsonrpc-primitives/src/types/split_storage.rs:48]: I'd even argue that `into_rpc_status_error` is more readable than just `into` :P

**Comment by jancionear** [comment]
> I can't find good documentation on that topic but perhaps @posvyatokum or @andrei-near will know.
TLDR
regular nodes only store the most recent chain history
archival nodes store full chain history
legacy archival nodes store everything in a single RocksDB, on a single, huge and expensive SSD disk
split store archival nodes store the most recent history and the rest of the history on two separate disks.
the hot disk is typically SSD, it's nice and fast, the cold disk is typically HDD, it's big and slow but cheap
benefit 1 - hot being small, is much faster and thanks to this the node can keep up with the network
benefit 2 - massive cost saving thanks to HDD being much cheaper, especially for such large volumes

Ahh, thanks, it starts making sense now :)

I think I saw something similar before, but it was done using something like `lvm-cache`/`bcachefs`. The SSD and HDD were connected into a single logical device, which used the SSD as a cache under the hood. All of the frequently accessed data was kept on the SSD automatically, while the cold data stayed on the HDD. The plus was that it didn't require any modifications in the application, but I have no idea if that would work with RocksDB.
